monitoring_scripts/featuredrift.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+from typing import List
+
+from evidently.report import Report
+from evidently.metrics import DatasetDriftMetric, DataDriftTable
+
+class FeatureDrift:
+    """
+    Calculates the per project feature drift of excerpts
+    Input: DataFrame with columns 'project_id', 'embeddings'
+    Output: DataFrame containing columns 'reference_project_id', 'current_project_id',
+            'reference_dataset_len', 'current_dataset_len', 'drift_share', 'number_of_columns',
+            'number_of_drifted_columns', 'share_of_drifted_columns', 'dataset_drift'
+    """
+    def __init__(self,
+        ref_df: pd.DataFrame,
+        cur_df: pd.DataFrame
+    ):
+        self.reference_df = ref_df
+        self.current_df = cur_df
+        
+        self.data_drift_dataset_report = Report(
+            metrics=[DatasetDriftMetric()]
+        )
+    
+    def _process_embeddings(self, embeddings: pd.Series) -> List[float]:
+        """
+        Reads the embeddings and gets the list
+        Input: Embedding Series
+        Output: Embedding List
+        """
+        return embeddings.apply(eval).apply(np.array).to_list()
+    
+    def _project_id_based_mask(self,
+        df: pd.DataFrame,
+        project_id: float=None
```

Comment:
Why do we have `project_id` of type float?

Reply:
fixed